### PR TITLE
fix: remove SDK registration and metrics `environment` field

### PR DESCRIPTION
**Pull request by irstavr**
## About the changes

We still hold environment field on SDK registration and metrics payloads despite being deprecated in v5. 

Here, we enforce the `environment` to be inferred from the API token rather than client-provided payload metadata.

Before:

```
resolveMetricsEnvironment(user, data) {
    if (user instanceof ApiUser) {
        if (user.environment !== ALL) {
            return user.environment;
        } else if (user.environment === ALL && data.environment) {
            return data.environment;    // <-- the change - this is GONE!
        }
    }
    return 'default';
}
```

Meaning that, when a wildcard-scoped token POSTs metrics with `environment` in the body, the server uses the body's value. But, now, the `environment` is not part of the schema, so, we remove the body field and metric attribution, thus, path changes to 'default' for those callers.

**Who's not affected**: everyone using narrow-scoped tokens (the already documented + recommended path, most customers do already).

**Who's affected**: users with instance-wide API tokens (environment='*'). The token-issuer doesn't fail, the metric just gets attributed to the wrong bucket. We already track these users and they are notified for the deprecation change. Thus, we assume no risk at the moment.

#### SDK situation 

The `unleash-node-sdk` already doesn't send `environment` in its [registration payload](https://github.com/Unleash/unleash-node-sdk/blob/747d20550595e08371a8bbb6f9825c5590588e55/src/metrics.ts#L414-#L426).
✔️  Node SDK users

But, even if any other SDK sends the `environment` field, the server silently drops it via Joi `stripUnknown: true`. 
✔️ No SDK breakage 

### In a nutshell, here, we:
- update OSS schemas (no environment params)
- remove the wildcard fallback on server logic part
- preserve-from-seenClients semantics (as broken test unveiled)
- adapt tests on new logic

Closes #EG-4386

## Discussion points

Update documentation with a note to explain that v8 attributes metrics from the token's environment, and customers using wildcard tokens for metrics need to issue narrower tokens or accept aggregation to default.

## OSS PR checklist

- [x] I have read and agree to the [Unleash Contributor License Agreement](https://github.com/Unleash/unleash/blob/main/CLA.md).
- [ ] I have added tests or explained why tests are not needed.
- [ ] I have updated documentation where relevant.

### CODE REVIEW

File: src/lib/features/metrics/client-metrics/metrics-service-v2.ts [hunk 1]
```diff
@@ -299,7 +300,7 @@ export default class ClientMetricsServiceV2 {
         ).map((name) => ({
             featureName: name,
             appName: value.appName,
-            environment: value.environment ?? 'default',
+            environment: environment ?? 'default',
```
Comment (sighphyre) [inline src/lib/features/metrics/client-metrics/metrics-service-v2.ts]: We're taking a non nullable string here, don't think we need the fallback anymore - which is nice because that fallback doesn't make sense anymore

```suggestion
            environment,
```

File: src/lib/features/metrics/client-metrics/metrics-service-v2.ts [hunk 2]
```diff
@@ -413,7 +414,7 @@ export default class ClientMetricsServiceV2 {
                             no: 0,
                             yes: 0,
                             appName,
-                            environment,
+                            environment: environment ?? 'default',
```
Comment (sighphyre) [inline src/lib/features/metrics/client-metrics/metrics-service-v2.ts]: Can't quite see why this one is necessary, can this now be null or are you just making the typing more honest?

File: src/lib/features/metrics/instance/instance-service.ts [hunk 3]
```diff
@@ -130,11 +128,16 @@ export default class ClientInstanceService {
...
         clientIp: string,
+        environment: string,
     ): Promise<void> {
         const value = await clientRegisterSchema.validateAsync(data);
         value.clientIp = clientIp;
         value.createdBy = SYSTEM_USER.username!;
         value.sdkType = 'backend';
+
+        const existing = this.seenClients[this.clientKey(value)];
+        value.environment = existing?.environment ?? environment; // existing or from the authenticated API token
```
Comment (sighphyre) [inline src/lib/features/metrics/instance/instance-service.ts:139]: This feels a little weird, what's the reasoning behind trying to get this from the existing seenClients? I think if we're passing in environment, we should just trust that to reduce mental overhead, unless there's an edge case I'm not seeing here

Comment (irstavr) [reply src/lib/features/metrics/instance/instance-service.ts:139]: there was[ a test breaking ](https://github.com/Unleash/unleash/blob/96e860a9814bb5726087e5dc8c0c8b9ecf4c0395/src/lib/features/metrics/instance/instance-service.test.ts#L322-#L356)expecting 'envs to be merged' - to reflect the behavior by preserving the 'from seenClients' semantics, I went for making the metric to func as a historical token state rather than current token state, seems it doesn't break a previous behavior - ( I did expect comments here :> )

Comment (sighphyre) [reply src/lib/features/metrics/instance/instance-service.ts:139]: Ah. Okay that test and the PR behind it seem to be pretty well reasoned through - it sounds like there are some ugly and subtle shenanigans in our protocol somewhere that I don't understand and it's safer to keep the behavior.

Cheers, makes sense!

**Review by sighphyre — APPROVED** [review]
Nice work here!
